### DISCUSSION TITLE
Replace controller

### DISCUSSION THREAD

**Discussion by Scotty42** [zwave-js/zwave-js-ui#2727] [Q&A] [answered]
Hi all, 
is there any preferred way to replace a controller without having to re-train the network and its nodes? 
In my scenario I presently use an older Razberry on some RPI and would replace the old Razberry with the latest 700 series chipset one. Looking forward to approaches and ideas. 
kind regards, Markus

**Comment by robertsLando** [comment]
AFAIK the best way to do this is to do an NVM backup of your controller, plug in the new one and do a NVM restore on it.

To be sure wait for @AlCalzone @blhoward2  or @kpine to confirm this

**Reply by AlCalzone** [reply]
The NVM backup way requires the old controller to be built using SDK 6.61 or newer. The alternative _Controller shift_ isn't supported atm.

I might try and figure out the memory format of older SDKs at some point though, since I also have an older Razberry in production

**Comment by kpine** [comment]
If you plan to go from a Razberry to another Razberry, I would contact Zwave.me support (either directly or in their own forums) and see if they support this kind of controller replacement, either through some tools or with their software. They use custom firmware for their devices, so they may have better accessibility.

**Comment by Scotty42** [comment]
The RaZerry 2016 controller reports FW 5.37 with SDK 6.2.0 - so NVM backup might not work as expected. But I'll ask in the Z-Wave.me forum and share their answer.

**Reply by kpine** [reply]
That's more than likely protocol v6.2 which is SDK 6.81.1 and that supports NVM backups. If the driver logs show that it supports SmartStart, then probably (?) supports NVM conversions as well. See issue https://github.com/zwave-js/node-zwave-js/issues/5160, feel free to add a startup log showing the controller interview if you can.

**Reply by kpine** [reply]
You can use Z-Way for sure:

- https://forum.z-wave.me/viewtopic.php?f=3422&t=35443
- https://forum.z-wave.me/viewtopic.php?f=3422&t=35053

But Z-Wave JS may work as well.

**Reply by AlCalzone** [reply]
If I remember correctly, the 2016 Razberry runs an old SDK, not 6.81

**Reply by kpine** [reply]
Yes, but the [firmware](https://z-wave.me/support/uzbrazberry-firmwares/) is upgrade-able. If OP has FW 5.37:

> 5.37: Updated SDK to 6.81.01. Fixed blink on sending and on SmartStart init. Reporting of incoming routes added in Z-Way Reoute Map.

OP will have to confirm though.

**Reply by Scotty42** [reply]
I'll have to download their Zway software and update the firmware then it seems. 
Their pre-built cmd line tool ZMEUpdate is not executable on my 64bit arch install. 
The mentioned zway backup/restore procedure likely is exactly the same NVM backup/restore. Nevertheless I'll try the firmware update first and report back on progress.
well, for the sake of z-way software I'll switch OS to a Debian image, do the controller works, and then we'll see

**Comment by Scotty42** [comment]
Using Z-Way 3.2.3 an update from 
`Hardware
Vendor:	RaZberry by Z-Wave.Me
Vendors Product ID:	1024 / 2
Z-Wave Chip:	ZW0500
Firmware
Library Type:	Static Controller
SDK Version:	6.81.01
Serial API Version:	05.37` to
`Hardware
Vendor:	RaZberry by Z-Wave.Me
Vendors Product ID:	1024 / 2
Z-Wave Chip:	ZW0500
Firmware
Library Type:	Static Controller
SDK Version:	6.82.01
Serial API Version:	05.39` 
was possible. Supported functions:
`NVMGetId (0x29) • NVMExtReadLongBuffer (0x2a) • NVMExtWriteLongBuffer (0x2b) • NVMExtReadLongByte (0x2c) • NVMExtWriteLongByte (0x2d)`

NVM backup from JS UI works with this version. Next step is to change boards and try 
1. NVM restore fro JS UI
2. and if that fails, restore Z-Way settings backup and Expert UI controller backup

**Comment by Scotty42** [comment, marked as answer]
Did replace the board and used Z-Way software to update firmware / boot loader to the latest versions, which now is:
`Hardware
Vendor:	RaZberry by Z-Wave.Me
Vendors Product ID:	1025 / 257
Z-Wave Chip:	ZW0700
Firmware
Library Type:	Static Controller
SDK Version:	7.17.00
Serial API Version:	07.21`

Afterwards killed Z-Way, restarted JS UI and restored NVM which **worked 👍 ** although Product and Product Code get overwritten by the old values.

Well, out of curiosity I restarted Z-Way server to see about Product Code and found there is another firmware to upgrade to 7.28. Which did overwrite the wrong Product and Product Code to their intended values. 
`Z-Wave Shield for Raspberry Pi | RaZberry 7 Pro`

The overall process might not be quick but 2-3 coffees and 2 hours of time allow to replace a controller without re-training nodes.
